mlfinlab/feature_importance/fingerpint.py

```diff
@@ -0,0 +1,259 @@
+        return store
+
+    def _get_non_linear_effect_estimation(self) -> dict:
+        """
+        Get non-linear effect estimates as as the mean absolute deviation of the total marginal (single variable) effect
+        around its corresponding linear effect.
+        :return: (dict) of non-linear effect estimates for each feature column
+        """
+        store = {}
+        for col in self.X.columns:
+            x = self.feature_values[col].reshape(-1, 1)
+            y = self.ind_partial_dep_functions[col]
+
+            lmodel = LinearRegression(fit_intercept=True, normalize=False)
+            lmodel.fit(x, y)
+            nonlinear_effect = np.mean(np.abs(lmodel.predict(x) - y.values))
+            store[col] = nonlinear_effect
+        return store
+
+    def get_pairwise_effect(self, combinations: list) -> dict:
+        """
+        Get pairwise effect estimates as the de-meaned joint partial prediction of the two variables minus the de-meaned
+        partial predictions of each variable independently.
+
+        :param combinations: (list) of tuples (feature_i, feature_j) to test pairwise effect
+        :return: None
+        """
+        store = {}
+
+        for pair in combinations:
+            function_values = []  # Array of pairwise interactions [f_k_l(not centered), f_k(centered), f_l(centered)]
+            col_k = pair[0]
+            col_l = pair[1]
+
+            y_cdf_k_centered = self.ind_partial_dep_functions[col_k] - np.mean(self.ind_partial_dep_functions[col_k])
+            y_cdf_l_centered = self.ind_partial_dep_functions[col_l] - np.mean(self.ind_partial_dep_functions[col_l])
+
+            for x_k, y_cdf_k in zip(self.feature_values[col_k], y_cdf_k_centered):
+                for x_l, y_cdf_l in zip(self.feature_values[col_l], y_cdf_l_centered):
+                    col_k_position = self.feature_column_position_mapping[col_k]
+                    col_l_position = self.feature_column_position_mapping[col_l]
+                    X_ = self.X.values.copy()
+                    X_[:, col_k_position] = x_k
+                    X_[:, col_l_position] = x_l
+
+                    y_pred = self._get_model_predictions(X_)
+                    y_cdf_k_l = y_pred.mean()
+
+                    function_values.append([y_cdf_k_l, y_cdf_k, y_cdf_l])
+
+            function_values = np.array(function_values)  # Convert to np.array to vectorize operations
+
+            # Cumulative pairwise effect value for a given feature
+            # Need to center f_k_l as f_k, f_l have been already centered
+            func_value = sum(abs((function_values[:, 0] - np.mean(function_values[:, 0]) - function_values[:,
+                                                                                                           1] - function_values[:, 2])))
+
+            store[str(pair)] = func_value / (self.num_values ** 2)
+
+        self.pair_wise_effect = {'raw': store, 'norm': _normalize(store)}
+
+    def fit(self) -> None:
```

Comment:
As per previous comments, lets move the computations from the constructor to this `.fit()` method.

Reply:
One more difficulty: we set-up a dataframe of individual dependence functions. The shape of this data frame is defined by X. When fit() method is called we need store these values, as they are used in pairwise_interaction effect

Reply:
Let me think how to make it both sklearn API like and beautiful

Reply:
Thats ok, the fit method can still call those methods which store the values in the class attributes?